### DISCUSSION TITLE
Any information, news or beta access to feature invalidating SSG pages with re-generating

### DISCUSSION THREAD

**Discussion by robertovg** [vercel/next.js#16112] [Help]
With the [Next.js 9.4](https://nextjs.org/blog/next-9-4#incremental-static-regeneration-beta) release there was some notes very interesting but I'm not able to find any more information about. The paragraph was the following:

> Up next, we will be working on a supplemental RFC to address two additional incremental static generation capabilities:
> 
> **Re-generating and invalidating multiple pages at once** (like your blog index and a certain blog post)
> Re-generating by listening to events (like CMS webhooks), ahead of user traffic

Thanks in advance 🙏

**Comment by robertovg** [comment]
Someone with information about if something new coming in Next.js 9.6?

**Comment by JuanmaMenendez** [comment]
Re-generating and invalidating multiple pages at once (like your blog index and a certain blog post) was already deployed.

What about "Re-generating by listening to events (like CMS webhooks), ahead of user traffic" ??

**Comment by bububo** [comment]
"Re-generating by listening to events (like CMS webhooks), ahead of user traffic" 
Looking forward this feature.
If someone knows a workaround how to trigger re-generating manually please share.

**Reply by JuanmaMenendez** [reply]
you just need to do a programmatically "get" request to the proper URL (eg: /productX) when your CMS Webhook notify your system that product X was changed

**Reply by bububo** [reply]
@JuanmaMenendez 
I think it will not work if there is _revalidate_ property. Imagine, we have _revalidate: 30_. That means that next js will not request data from the API more then once per minute. And our get request may not trigger the next js request for new data.

**Reply by JuanmaMenendez** [reply]
In that case, you need to set _revalidate: 30_ and then do the get request when something changes in the CMS. Anyway this is just a temporal workaround until they release the functionality to officially trigger the revalidation

**Comment by michrome** [comment]
Cross linking this to #16488 which is a request for the same: regeneration based on an event, to cut short the wait for a revalidation period.
